### DISCUSSION TITLE
Passports and Badge for 2FA auth

### DISCUSSION THREAD

**Discussion by Richard87** [symfony/symfony#43189] [Q&A] [answered]
Hi!

In Symfony Guard I'm using attributes to say a user is logged in with 2FA, so the user gets extra privileges.

A passport with Badges sounds like a better way to do it, but is it?

I want to expire the badge after a timeout, and if the user is authenticated, trying to access a page requiring 2FA, then redirecting the user to the login-page instead. 

Are there any recommendations on how to do this? Would badges help me, or is attributes equally simple?

**Comment by wouterj** [comment, marked as answer]
Passports and badges are only used during authentication (e.g. before a user is considered "authenticated" and before a token is created). After authentication is completed, the token will be created and saved to the session.

It seems like what you want is to do something after authentication (checking expiry), which is something you still need to do in the token.

**Reply by Richard87** [reply]
Thanks for the explanation!

Good to know, then I will continue to use token attributes! :)
